website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,96 @@
+## Usage
+
+### When to use
+
+- When displaying code examples and longer snippets of code that benefit from syntax highlighting.
+
+### When not to use
+
+- As a full-blown code editor.
+
+### When to use a CodeBlock vs a CopySnippet
+
+There is some overlap in the copying functionality of the CopySnippet and the CodeBlock. Which to use generally comes down to the complexity of the code/value displayed within the component, and whether the user benefits from seeing the larger context of the code example.
+
+Use a CodeBlock if:
+
+- If viewing or displaying code is the primary purpose, and copying the code is secondary.
+- If the example consists of a command, e.g., a `curl` and `bash` script. These are oftentimes on a single line, but consist of multiple commands and functions.
+
+Use a CopySnippet if:
+
+- If copying the code is the primary purpose, and viewing the code is secondary.
+- Allowing the user to copy an API key or other **single** value or string.
+- If seeing the value in a larger context of where it's expected to be pasted into isn’t necessary.
+
+## Standalone
+
+The `isStandalone` property increases the portability of the CodeBlock to ensure that it can be used in different contexts. For example, a common use case of the CodeBlock is in a "standalone" context, which can be part of a form, multi-step process, and is generally a part of the normal layout flow.
+
+![Code Block with rounded corners in a standalone context](/assets/components/code-block/code-block-rounded-standalone.png)
+
+Sometimes it may be necessary to use the CodeBlock in a more dense layout or nested within another component. In this circumstance, setting `isStandalone` to false ensures that the CodeBlock fits alongside other elements, in a split view, or as part of a larger layout mechanism.
+
+![Code Block in a block context](/assets/components/code-block/code-block-block-level.png)
+
+## Header
+
+Use a `title` or `description` in the header to provide additional information, instructions, or to label a CodeBlock. Both of these properties are optional, but including them can help to provide additional context about a specific block of code.
+
+![Example metadata in the Code Block](/assets/components/code-block/code-block-metadata.png)
+
+### When not to use a header
+
+There can be an overlap between content that you may choose to include in the header as a `title` or `description`, and content that is part of the normal layout flow in a headline or paragraph. If it is necessary to elevate this content in the hierarchy of the page, we recommend including it in the normal layout flow, rather than as a `title` or `description` within the Code Block.
```

Comment:
```suggestion
There can be an overlap between content that you may choose to include in the header as a `title` or `description`, and content that is part of the normal layout flow in a headline or paragraph. If it is necessary to elevate this content in the hierarchy of the page, we recommend including it in the normal layout flow, rather than as a `title` or `description` within the CodeBlock.
```